composer/loggers/progress_bar_logger.py

```diff
@@ -21,44 +21,67 @@
 _IS_TRAIN_TO_KEYS_TO_LOG = {True: ['loss/train'], False: ['metrics/eval/Accuracy']}
 
 
-@dataclass
-class _ProgressBarLoggerInstanceState:
-    total: Optional[int]
-    description: str
-    position: int
-    keys_to_log: List[str]
-    n: int
-    epoch_metrics: Dict[str, Any]
+class _ProgressBar:
 
-
-class _ProgressBarLoggerInstance:
-
-    def __init__(self, file: TextIO, state: _ProgressBarLoggerInstanceState) -> None:
-        self.state = state
+    def __init__(
+        self,
+        total: Optional[int],
+        position: int,
+        bar_format: str,
+        file: TextIO,
+        metrics: Dict[str, Any],
+        keys_to_log: List[str],
+        unit: TimeUnit = TimeUnit.EPOCH,
+        epoch_style: bool = False,
```

Comment:
Curious if we need the `epoch_style` argument? For evaluation, `batch_in_epoch` is always the same as `batch`, (same thing with samples and tokens), so I think it should be safe to always use the `x_in_epoch` value? 

I realize that `Timestamp.get(TimeUnit)` has no way to get the `x_in_epoch` -- that's an oversight. Thoughts about adding in a `get_in_epoch(TimeUnit)` helper?

Reply:
`epoch_style` is also used during training when the `unit == TimeUnit.EPOCH`, so this is needed to switch that exception.